os/sys/rtimer.h

```diff
@@ -163,6 +164,16 @@ enum {
 int rtimer_set(struct rtimer *task, rtimer_clock_t time,
 	       rtimer_clock_t duration, rtimer_callback_t func, void *ptr);
 
+/**
+ * \brief      Schedules a real-time task for a given instant
+ * \param task metadata about the real-time task
+ *
+ *             This variant does not auto-delay real-time tasks that
+ *             lie less than RTIMER_GUARD_TIME ticks in the future,
+ *             but returns an error code in such cases.
+ */
+int rtimer_set_precise(struct rtimer *task);
```

Comment:
I only had a quick look, can the pointer parameters be const?

Reply:
Thank you for pointing this out. You can, however, ignore the rtimer commits. I will separate them if there is interest in finalizing this PR.

Reply:
I have removed the rtimer commits from this PR.